### PR TITLE
sdi-361 Adding framework for Incentive mapping support. Further PRs t…

### PR DESCRIPTION
**Pull request by lbennett-moj**
…o come

### CODE REVIEW

File: src/main/resources/db/migration/V1_10__incentive.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    nomis_incentive_sequence       bigint       not null,
+    incentive_id                   bigint       not null  PRIMARY KEY,
+    label                          varchar(20),
+    mapping_type                   varchar(20)  not null,
+    when_created                   timestamp with time zone not null default now(),
+    constraint incentive_mapping_nomis_id_unique  unique(nomis_booking_id, nomis_incentive_sequence)
+);
+
+create index incentive_mapping_label_index on incentive_mapping (label);
+create index incentive_mapping_when_created_index on incentive_mapping (when_created);
```
Comment (steverendell) [inline src/main/resources/db/migration/V1_10__incentive.sql:13]: You could probably do with an index on nomis_booking_id, nomis_incentive_sequence to support findOneByNomisBookingIdAndNomisIncentiveSequence()

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/helper/builders/IncentiveRepository.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.helper.builders
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Repository
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.jpa.repository.IncentiveMappingRepository
+
+@Repository
+@Transactional
+class IncentiveRepository(@Qualifier("incentiveMappingRepository") private val incentiveRepository: IncentiveMappingRepository) : IncentiveMappingRepository by incentiveRepository
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/helper/builders/IncentiveRepository.kt:10]: Not sure what this does .. is it referenced anywhere?

**Review by steverendell — APPROVED** [review]
lgtm with a couple of comments

File: src/main/resources/db/migration/V1_10__incentive.sql (hunk 1 above)
Comment (lbennett-moj) [reply src/main/resources/db/migration/V1_10__incentive.sql:13]: I have got one in the flyway script

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/helper/builders/IncentiveRepository.kt (hunk 2 above)
Comment (lbennett-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/helper/builders/IncentiveRepository.kt:10]: There are two IncentiveMappingRepository beans and unless I put a qualifier in - I got a compilation error. I'm sure there must be a way around it and will have another look

**Pull request merged**